internal/devbox/devbox.go

```diff
@@ -1039,11 +1039,14 @@ func (d *Devbox) flakeDir() string {
 	return filepath.Join(d.projectDir, ".devbox/gen/flake")
 }
 
-// ConfigPackageNames returns the package names as defined in devbox.json
-func (d *Devbox) PackageNames() []string {
-	// TODO savil: centralize implementation by calling d.configPackages and getting pkg.Raw
-	// Skipping for now to avoid propagating the error value.
-	return d.cfg.PackagesVersionedNames()
+// AllPackageNames returns the all package names, including those added
+// by plugins
+func (d *Devbox) AllPackageNames() []string {
+	result := make([]string, 0, len(d.cfg.Packages()))
+	for _, p := range d.cfg.Packages() {
+		result = append(result, p.VersionedName())
+	}
+	return result
```

Comment:
Are the names from `AllPackageNames` exactly matching the set of packages in `AllPackages` functions below?

`AllPackages` seems to respect `__remove_trigger_package`. Does `AllPackageNames` do that as well?

Reply:
Good catch. Not sure. I'll audit.

Reply:
Definitely needed in most places.  Added back in. I think we can improve on this (some callers want different representations of this list) but for now the safest fix is to just include everything (top level, plugin packages, and also trigger removed)